inference-engine/tests/functional/plugin/gna/pass_tests/padded2valid_conv.cpp

```diff
@@ -0,0 +1,307 @@
+        auto filter_size = std::accumulate(std::begin(kernel), std::end(kernel), 1, std::multiplies<size_t>());
+        auto filter_weights = CommonTestUtils::generate_float_numbers(numOutChannels * inputShape[3] * filter_size, -0.5f, 0.5f);
+        auto conv = builder::makeConvolution(transpose_in, ngPrc, kernel, stride, padBegin,
+            padEnd, dilation, padType, numOutChannels, false, filter_weights);
+        auto transpose_out_order = op::Constant::create(element::i64, Shape{ 4 }, { 0, 2, 3, 1 });
+        auto bias_const = builder::makeConstant(ngPrc, bias_shape, bias_weights, true);
+        std::shared_ptr<Node> last_op = std::make_shared<Transpose>(conv, transpose_out_order);;
+
+        switch (model) {
+        case modelType::TranspConvBcastAddTransp:
+        {
+            auto bias = std::make_shared<Add>(conv, bias_const);
+            last_op = std::make_shared<Transpose>(bias, transpose_out_order);
+        }
+        break;
+
+        case modelType::TranspConvBcastAddMaxPoolTransp:
+        {
+            auto bcast_add = std::make_shared<Add>(conv, bias_const);
+            auto maxpool = std::make_shared<MaxPool>(bcast_add, Strides{ 1, 1 }, Shape{ 0, 0 }, Shape{ 0, 0 }, maxpool_shape);
+            last_op = std::make_shared<Transpose>(maxpool, transpose_out_order);
+        }
+        break;
+
+        case modelType::TranspConvBcastAddActTransp:
+        {
+            auto bcast_add = std::make_shared<Add>(conv, bias_const);
+            auto activation = std::make_shared<Relu>(bcast_add);
+            last_op = std::make_shared<Transpose>(activation, transpose_out_order);
+        }
+        break;
+
+        case modelType::TranspConvBcastAddMaxPoolActTransp:
+        {
+            auto bcast_add = std::make_shared<Add>(conv, bias_const);
+            auto max_pool = std::make_shared<MaxPool>(bcast_add, Strides{ 1, 1 }, Shape{ 0, 0 }, Shape{ 0, 0 }, maxpool_shape);
+            auto activation = std::make_shared<Relu>(max_pool);
+            last_op = std::make_shared<Transpose>(activation, transpose_out_order);
+        }
+        break;
+
+        case modelType::TranspConvTranspBcastAdd:
+        {
+            bias_const = std::make_shared<Constant>(ngPrc, transp_bias_shape);
+            last_op = std::make_shared<Add>(last_op, bias_const);
+        }
+        break;
+
+        case modelType::TranspConvTranspBcastAddAct:
+        {
+            bias_const = builder::makeConstant(ngPrc, transp_bias_shape, bias_weights, true);
+            auto bcast_add = std::make_shared<Add>(last_op, bias_const);
+            last_op = std::make_shared<Relu>(bcast_add);
+        }
+        break;
+
+        case modelType::TranspConvTransp:
+        default:
+            break;
+        }
+
+        function = std::make_shared<Function>(NodeVector{ last_op }, ParameterVector{ input });
+    }
+};
+
+class GnaPadded2ValidConvTest : public Padded2ValidConvTest, GnaLayerTestCheck {
```

Comment:
What is the difference between GnaPadded2ValidConvTest and Padded2ValidConvTest? Does the first test 2d  convolution while the second is for 1d? It's not qute clear from their names, it looks like the second one isn't related to GNA.

Reply:
GnaPadded2ValidConvTest is wrapped with a check for GNA library version 2.1, but I'll change the name to reflect that it's only for conv2d.